Prevent logging from exiting on empty lines

schultetwin1 opened the pull request:
Before this commit, `freadline` did not distinguish between an empty
line and EOF. This would cause a printf such as

```c
printf("\nHello, World\n");
```

to cause dlog, dslog, or drlog to exit. This change attempts to
separate the EOF case and the empty line case to prevent this logic
from occurring.

Fixes #10
Fixes #12

aperezdc commented on util.h:
```diff
@@ -21,6 +21,8 @@
 
 #define ERRSTR (strerror (errno))
 
+#define DMON_EOF (-2)
```
Using this constant looks clunky to me. End-of-file is not technically an error, that's why zero is returned in that case, and the code here is using the POSIX convention of returning a negative number to indicate an error—it could be any negative number. But the function did in fact read one byte (the delimiter) so returning zero (which means “end of file”) is not quite right.

A better approach would be to make `freaduntil()` include the delimiter in the resulting buffer, and if the returned buffer has `dbuf_size(&linebuf) == 1` then that means that only the delimiter has been put in the buffer. As a bonus, including the new line in the buffer means that it does not need to be added later to the output by hand.

schultetwin1 replied:
> A better approach would be to make...

Great suggestion. I like this solution a lot better! I have updated this PR with that changes. This PR will now fix #10 and #11.

aperezdc reviewed (changes requested):
@schultetwin1 Good catch, thanks a lot!

Indeed the logging tools should be more robust and handle empty input lines correctly. I think there is a better way of taking them into account. It would be great if you could try the approach that I suggest below.

Thanks in advance! :muscle:

Commit:
Prevent logging from exiting on empty lines

Before this commit, `freadline` did not distinguish between an empty
line and EOF. This would case a printf such as

```
printf("\nHello, World\n");
```

to cause dlog, dslog, or drlog to exit.

In order to fix this, this commit includes the delimiter in the result
of `freaduntil`, which means `\n` will be included in all calls to
`freadline`.

Fixes #10
Fixes #12

aperezdc reviewed (approved):
That was quick, thanks! The changes look good.
